packages/rest/src/__tests__/acceptance/caching-interceptor/caching-interceptor.acceptance.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {intercept} from '@loopback/context';
+import {get, param} from '@loopback/openapi-v3';
+import {
+  Client,
+  createRestAppClient,
+  expect,
+  givenHttpServerConfig,
+} from '@loopback/testlab';
+import {RestApplication} from '../../..';
+import {
+  cache,
+  cachedResults,
+  CachingInterceptorProvider,
+  clearCache,
+  status,
+} from './caching-interceptor';
+
+describe('caching interceptor', () => {
+  let client: Client;
+  let app: RestApplication;
+
+  before(givenAClient);
+  after(async () => {
+    await app.stop();
+  });
+
+  context('toUpperCase with bound caching interceptor', () => {
+    it('invokes the controller method if not cached', async () => {
+      await client.get('/toUpperCase/Hello').expect(200, 'HELLO');
+      expect(status.returnFromCache).to.be.false();
+    });
+
+    it('returns from cache without invoking the controller method', async () => {
+      for (let i = 0; i <= 5; i++) {
+        await client.get('/toUpperCase/Hello').expect(200, 'HELLO');
```

Comment:
What happens when I run this test in isolation and  don't execute `it('invokes the controller method if not cached'` first? Will it still pass?